Source/Orts.Simulation/Simulation/Activity.cs

```diff
@@ -1502,17 +1518,23 @@ override public Boolean Triggered(Activity activity)
         {
             var triggered = false;
             var e = this.ParsedObject as Orts.Formats.Msts.EventCategoryLocation;
+            Train train = Simulator.PlayerLocomotive.Train;
+            if (ParsedObject.TrainService != "" && Train != null)
+            {
+                if (Train.FrontTDBTraveller != null) train = Train;
+                else return triggered;
```

Comment:
Swapping this around so the return is first might make this more readable, as you'll not need the `else`.

Reply:
Will do.